src/interpreter.rs

```diff
@@ -83,12 +79,35 @@ impl Scanner<'_> {
             b"+" => Plus,
             b";" => Semicolon,
             b"*" => Star,
-            b"!" => decide_token_type(Bang, (BangEqual, b"="), self.next_byte()),
-            b"=" => decide_token_type(Equal, (EqualEqual, b"="), self.next_byte()),
-            b">" => decide_token_type(Greater, (GreaterEqual, b"="), self.next_byte()),
-            b"<" => decide_token_type(Less, (LessEqual, b"="), self.next_byte()),
-            b"/" => decide_token_type(Slash, (SlashSlash, b"/"), self.next_byte()),
+            b"!" => self.decide_token_type(Bang, (BangEqual, b"=")),
+            b"=" => self.decide_token_type(Equal, (EqualEqual, b"=")),
+            b">" => self.decide_token_type(Greater, (GreaterEqual, b"=")),
+            b"<" => self.decide_token_type(Less, (LessEqual, b"=")),
+            b"/" => self.decide_token_type(Slash, (SlashSlash, b"/")),
             [digit] if digit.is_ascii_digit() => self.treat_number(),
+            &[a] if a.is_ascii_alphabetic() || a == b'_' => {
+                let range = self.measure_word();
+                let word = &self.bytes[range];
+                match word {
+                    b"and" => Keyword(token::Keyword::And),
+                    b"class" => Keyword(token::Keyword::Class),
+                    b"else" => Keyword(token::Keyword::Else),
+                    b"false" => Keyword(token::Keyword::False),
+                    b"for" => Keyword(token::Keyword::For),
+                    b"fun" => Keyword(token::Keyword::Fun),
+                    b"if" => Keyword(token::Keyword::If),
+                    b"nil" => Keyword(token::Keyword::Nil),
+                    b"or" => Keyword(token::Keyword::Or),
+                    b"print" => Keyword(token::Keyword::Print),
+                    b"return" => Keyword(token::Keyword::Return),
+                    b"super" => Keyword(token::Keyword::Super),
+                    b"this" => Keyword(token::Keyword::This),
+                    b"true" => Keyword(token::Keyword::True),
+                    b"var" => Keyword(token::Keyword::Var),
+                    b"while" => Keyword(token::Keyword::While),
+                    bytes => Type::Identifier(String::from_utf8(bytes.to_vec()).unwrap()),
+                }
+            }
```

Comment:
This part would be better if extracted into a function.

Reply:
Also, adding a specific test for number literals to detect prevent that bug would be better than testing against it under `identifier`.